aodntools/timeseries_products/aggregated_timeseries.py

```diff
@@ -1,113 +1,134 @@
         error_list.append('Wrong file version: ' + nc_file_version)
 
-    if 'TIME' not in variables:
-        error_list.append('TIME variable missing')
+    if VoI not in file_variables:
+        error_list.append(VoI + ' not in file')
+    else:
+        for dimension in list(nc[VoI].dims):
+            if dimension not in allowed_dimensions:
+                error_list.append(dimension+' is not an allowed dimension for ' + VoI)
 
-    if 'LATITUDE' not in variables:
-        error_list.append('LATITUDE variable missing')
+    if 'NOMINAL_DEPTH' not in file_variables and 'instrument_nominal_depth' not in attributes:
+        error_list.append('no NOMINAL_DEPTH')
 
-    if 'LONGITUDE' not in variables:
-        error_list.append('LONGITUDE variable missing')
+    return error_list
 
-    if 'NOMINAL_DEPTH' not in variables and 'instrument_nominal_depth' not in attributes:
-        error_list.append('no NOMINAL_DEPTH')
 
-    if VoI not in variables:
-        error_list.append(VoI + ' not in file')
+def get_variable_values(nc, variable):
+    """
+    Get value sof the variable and its QC flags.
+    If variable is not present, nan returned
+    If variable present but not its QC flags, QC set to 9
+    :param nc: dataset
+    :param variable: name of the variable to get
+    :return: variable values and variable qc flags
+    """
+    n_records = len(nc.TIME)
+    file_variables = list(nc.variables)
+
+    if variable in file_variables:
+        variable_values = nc[variable]
+        if variable+'_quality_control' in file_variables:
+            variableQC_values = nc[variable+'_quality_control']
+        else:
+            variableQC_values = np.repeat(9, n_records)
     else:
-        VoIdimensions = list(nc[VoI].dims)
-        if 'TIME' not in VoIdimensions:
-            error_list.append('TIME is not a dimension')
-        if 'LATITUDE' in VoIdimensions and len(nc.LATITUDE) > 1:
-            error_list.append('more than one LATITUDE')
-        if 'LONGITUDE' in VoIdimensions and len(nc.LONGITUDE) > 1:
-            error_list.append('more than one LONGITUDE')
-        for d in range(len(VoIdimensions)):
-            if VoIdimensions[d] not in allowed_dimensions:
-                error_list.append('not allowed dimensions: ' + VoIdimensions[d])
-                break
-        # test the units. not used. To be implemented in the future
-        # if nc[VoI].attrs['units'] != variable_attribute_dictionary[VoI]['units']:
-        #     error_list.append('Wrong units: ' + nc[VoI].attrs['units'])
+        variable_values = np.repeat(np.nan, n_records)
+        variableQC_values = np.repeat(np.nan, n_records)
+
+    return variable_values, variableQC_values
+
+
+def get_varvalues(nc, varname):
+    """
+    Return a 1D array of 2D values
+    :param nc: dataset
+    :param varname: Variable of interest
+    :return: variable values flattened
+    """
+    return nc[varname].values.flatten()
+
+
+def get_instrumentID(nc):
```

Comment:
Should be `get_instrument_id` by convention

Reply:
fixed